core/rawdb/scroll.go

```diff
@@ -0,0 +1,166 @@
+package rawdb
+
+import (
+	"bytes"
+	"encoding/binary"
+	"math/big"
+
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/core/types"
+	"github.com/scroll-tech/go-ethereum/ethdb"
+	"github.com/scroll-tech/go-ethereum/log"
+	"github.com/scroll-tech/go-ethereum/rlp"
+)
+
+//
+func WriteSyncedL1BlockNumber(db ethdb.KeyValueWriter, blockNumber *big.Int) {
+	value := []byte{0}
+	if blockNumber != nil {
+		value = blockNumber.Bytes()
+	}
+	if err := db.Put(syncedL1BlockNumberKey, value); err != nil {
+		log.Crit("Failed to synced L1 block number", "err", err)
+	}
+}
+
+//
+func ReadSyncedL1BlockNumber(db ethdb.Reader) *big.Int {
+	data, _ := db.Get(syncedL1BlockNumberKey)
+	if len(data) == 0 {
+		return nil
+	}
+	return new(big.Int).SetBytes(data)
+}
+
+//
+func WriteL1Message(db ethdb.KeyValueWriter, msg *types.L1MessageTx) {
+	bytes, err := rlp.EncodeToBytes(msg)
+	if err != nil {
+		log.Crit("Failed to RLP encode L1 message", "err", err)
+	}
+	enqueueIndex := msg.Nonce
+	if err := db.Put(L1MessageKey(enqueueIndex), bytes); err != nil {
+		log.Crit("Failed to store L1 message", "err", err)
+	}
+}
+
+//
+// TODO: consider writing messages in batches
+func WriteL1Messages(db ethdb.KeyValueWriter, msgs []types.L1MessageTx) {
+	for _, msg := range msgs {
+		WriteL1Message(db, &msg)
+	}
+}
+
+//
+func ReadL1MessageRLP(db ethdb.Reader, enqueueIndex uint64) rlp.RawValue {
+	data, err := db.Get(L1MessageKey(enqueueIndex))
+	if err != nil {
+		log.Crit("Failed to load L1 message", "enqueueIndex", enqueueIndex, "err", err)
+	}
+	return data
+}
+
+//
+func ReadL1Message(db ethdb.Reader, enqueueIndex uint64) *types.L1MessageTx {
+	data := ReadL1MessageRLP(db, enqueueIndex)
```

Comment:
What will happen when `data` has nonzero length and `err` is not `nil` when call `ReadL1MessageRLP`.

Reply:
`log.Crit` will terminate the node.